lib/rspec/support/spec/shell_out.rb

```diff
@@ -16,26 +16,9 @@ def with_env(vars)
         end
       end
 
-      if Open3.respond_to?(:capture3) # 1.9+
-        def shell_out(*command)
-          stdout, stderr, status = Open3.capture3(*command)
```

Comment:
I'm not sure how this used to work.
jruby-9.1 (9.1.17.0):
```
> require 'open3'
 => true
jruby-9.1.17.0 :003 > Open3.respond_to?(:capture3)
 => true
jruby-9.1.17.0 :004 > o,e,s=Open3.capture3("ruby -e 'warn \"lol\"'")
lol
 => ["", "", #<Process::Status: pid 12445 exit 0>]
```

jruby-9.2 (9.2.13.0), MRI 2.7.1:
```
> require 'open3'
=> true
[2] pry(main)> o,e,s=Open3.capture3("ruby -e 'warn \"lol\"'")
=> ["", "lol\n", #<Process::Status: pid 12883 exit 0>]
```

Can't find anything related to `capture3` in JRuby's changelog.
I was going to add a check for JRuby < 9.2 and use `popen3`, but:
```
jruby-9.1.17.0 :006 > Open3.popen3("ruby -e 'puts \"lol\"'") do |_,o,e|
jruby-9.1.17.0 :007 >     puts "ERR: #{e.read}", "OUT: #{o.read}"
jruby-9.1.17.0 :008?>   end
lol
ERR:
OUT:
 => nil
```

also, there were two `popen3` related issues fixed in 9.1 and 9.2:
#5235 - popen3 doesn't works fine with java 9/10 and jruby > 1.9.13
#3948 - Open3.popen3 with block fails on Windows
We can't discard Java 9/10 and Windows, can we?

Locally, I'm getting this with 9.1.17.0:
```
  1) rspec warnings and deprecations works when required in isolation
     Failure/Error: DEFAULT_FAILURE_NOTIFIER = lambda { |failure, _opts| raise failure }
       expected "" to start with "DEPRECATION: foo is deprecated"

Failed examples:

rspec ./spec/rspec/support/warnings_spec.rb:11
```
This also fails in [GHA](https://github.com/rspec/rspec-support/runs/1438630956?check_suite_focus=true), and locally on `4-0-dev`, and `main`, with both my changes to other repos and other RSpec gems from RubyGems.

`USE_GIT_REPOS=1` doesn't work with Bunlder 2:
```
NoMethodError: undefined method `[]' for nil:NilClass
  /Users/pirj/.rvm/rubies/jruby-9.1.17.0/lib/ruby/gems/shared/gems/bundler-2.1.4/lib/bundler/source/git/git_proxy.rb:90:in `version'
  /Users/pirj/.rvm/rubies/jruby-9.1.17.0/lib/ruby/gems/shared/gems/bundler-2.1.4/lib/bundler/source/git/git_proxy.rb:147:in `block in copy_to'
  org/jruby/RubyDir.java:323:in `chdir'
```

Yet, still, with Bundler 1 and `rspec-*` from repos, the failure remains.

So, on `main`, with git repos, on JRuby 9.1.17.0, I'm getting:
```
520 examples, 3 failures, 5 pending

Failed examples:

rspec ./spec/rspec/support/spec/shell_out_spec.rb:6 # RSpec::Support::ShellOut shells out and returns stdout and stderr
rspec ./spec/rspec/support/spec/shell_out_spec.rb:33 # RSpec::Support::ShellOut passes along the provided ruby flags
rspec ./spec/rspec/support/warnings_spec.rb:11 # rspec warnings and deprecations works when required in isolation
```

which are exactly the same failures I'm getting with my changes on CI.

Is this something you've seen along your journey with GHA, @JonRowe, @benoittgt ?

Reply:
Just in case:
```
$ ruby --version
jruby 9.1.17.0 (2.3.3) 2018-04-20 d8b1ff9 OpenJDK 64-Bit Server VM 14.0.1+14 on 14.0.1+14 +jit [darwin-x86_64]
```

Reply:
Er. Pass.